app/desmos/cmd/root.go

```diff
@@ -23,58 +20,49 @@ import (
 	"github.com/cosmos/cosmos-sdk/x/auth/tx"
 	txmodule "github.com/cosmos/cosmos-sdk/x/auth/tx/config"
 	"github.com/cosmos/cosmos-sdk/x/auth/types"
-
-	"github.com/desmos-labs/desmos/v6/app"
+	"github.com/cosmos/cosmos-sdk/x/genutil"
+	genutiltypes "github.com/cosmos/cosmos-sdk/x/genutil/types"
+	"github.com/cosmos/cosmos-sdk/x/gov"
+	govclient "github.com/cosmos/cosmos-sdk/x/gov/client"
+	govtypes "github.com/cosmos/cosmos-sdk/x/gov/types"
+	paramsclient "github.com/cosmos/cosmos-sdk/x/params/client"
 )
 
-// NewRootCmd creates a new root command for desmos. It is called once in the
+// NewRootCmd creates a new root command for simd. It is called once in the
 // main function.
 func NewRootCmd() *cobra.Command {
 	// Read in the configuration file for the sdk
 	cfg := sdk.GetConfig()
 	app.SetupConfig(cfg)
 	cfg.Seal()
 
-	var (
-		interfaceRegistry  codectypes.InterfaceRegistry
-		appCodec           codec.Codec
-		txConfig           client.TxConfig
-		autoCliOpts        autocli.AppOptions
-		moduleBasicManager module.BasicManager
-		clientCtx          *client.Context
-	)
-
-	if err := depinject.Inject(
-		depinject.Configs(
-			app.GetAppConfig(),
-			depinject.Supply(
-				log.NewNopLogger(),
-				simtestutil.NewAppOptionsWithFlagHome(tempDir()),
-			),
-			depinject.Provide(
-				ProvideClientContext,
-				ProvideKeyring,
-			),
-		),
-		&interfaceRegistry,
-		&appCodec,
-		&txConfig,
-		&autoCliOpts,
-		&moduleBasicManager,
-		&clientCtx,
-	); err != nil {
-		panic(err)
+	// we "pre"-instantiate the application for getting the injected/configured encoding configuration
+	tempApp := app.NewDesmosApp(log.NewNopLogger(), dbm.NewMemDB(), nil, true, simtestutil.NewAppOptionsWithFlagHome(tempDir()))
+	encodingConfig := params.EncodingConfig{
+		InterfaceRegistry: tempApp.InterfaceRegistry(),
+		Codec:             tempApp.AppCodec(),
+		TxConfig:          tempApp.TxConfig(),
+		Amino:             tempApp.LegacyAmino(),
```

Comment:
We should set command manually instead of depinject since Desmos has non app wiring modules (ibc and wasm).
See: https://docs.cosmos.network/main/build/building-apps/app-go-v2#registering-non-app-wiring-modules